### PR TITLE
EAB-628 Added support for a defaultValue prop on the DateInput component

### PR DESCRIPTION
**Pull request by LewisClark-HO**
### Description
Added support for a default value to be supplied in a Form Renderer JSON file for any DateInput component. Any date string in the format `DD-MM-YYYY` is supported. Using the word `today` instead of a date string will cause Form Renderer to supply today's date, in the form `DD-MM-YYYY`, as the default value.

### To test
Related unit tests have been added. To test this functionality, define a date component in the Form Renderer sandbox and add the field:
```
...
"defaultValue": "01-01-2022",
...
```
or
```
...
"defaultValue": "today",
...
```

### CODE REVIEW

**Comment by LewisClark-HO** [comment]
Would appreciate any tips on a sane way to avoid the GitHub warning - I thought using an empty dependency array for a `useEffect` is the correct way of ensuring it only runs once?

File: src/DateInput/DateInput.jsx [hunk 1]
```diff
@@ -79,6 +80,15 @@ const DateInput = ({
...
   }, [dateChanged, date, fieldId, onChange, setDateChanged]);
 
+  useEffect(() => {
+    if (!value && defaultValue) {
+      setDate(toDateFromString(defaultValue));
+      if (typeof onChange === 'function') {
+        onChange({ target: { name: fieldId, value: defaultValue }});
+      }
+    }
+  }, []);
```
Comment (paulfgraham) [inline src/DateInput/DateInput.jsx:90]: As GitHub points out, this dependency array is needed.

File: src/DateInput/DateInput.jsx [hunk 2]
```diff
@@ -79,6 +80,15 @@ const DateInput = ({
     }
   }, [dateChanged, date, fieldId, onChange, setDateChanged]);
 
+  useEffect(() => {
+    if (!value && defaultValue) {
```
Comment (paulfgraham) [inline src/DateInput/DateInput.jsx:84]: If the field gets cleared out, it's going to go back to the default value, rather than remaining cleared out.

**Review by paulfgraham — CHANGES_REQUESTED** [review]
I don't think this is going to behave the way you expect. Every time you load up the form, it's going to set it to show the `defaultValue` if the field hasn't been populated, which doesn't seem right.

Also, I don't see why you're using a separate property to the `value` property - it doesn't add anything and simply makes this component more complicated as you now have to track both and, as stated, they're not going to play nicely the way you have the code set up.

**Pull request closed without merging**